### PR TITLE
#1792 Edit prescription line quantities

### LINKED ISSUE #1792: Edit quantities of prescription lines
## Is your feature request related to a problem? Please describe.

Need to be able to edit quantities of trans lines for a prescription

## Describe the solution you'd like

Hook the callback up for the stepper component

## Implementation

N/A

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
### BRANCHED FROM #1782 

Fixes #1792 

## Change summary
- Adds a prescription action for updating quantities
- Adds pass through of a dispatcher for the added action to the stepper of each row
- Adds some state to the stepper as having this update through the reducer and all the components didn't have a good feel, it was slightly too slow to have a responsive UI for updates every 50ms. Possibly this is why you don't see this kind of feature too often! Anyway, state has been added to update itself, while still passing the value back, just not causing a state update at the parent to propagate down. Causes more confusing code, but worth it for this exception?

editQuantities

## Testing

N/A

### Related areas to think about

N/A

### CODE REVIEW

File: src/widgets/StepperInput.js [hunk 1]
```diff
@@ -23,26 +26,33 @@ import { AddIcon, MinusIcon } from './icons';
  *
  */
 export const StepperInput = ({ value, onChangeText, lowerLimit, upperLimit, textInputStyle }) => {
+  const [, setCurrentValueState] = React.useState(Number(value));
   const currentValue = React.useRef(Number(value));
   const currentAdjustmentAmount = React.useRef(1);
   const valueAdjustmentInterval = React.useRef();
   const adjustmentIncreaseInterval = React.useRef();
 
+  const parseNumber = number => Math.max(Math.min(number, upperLimit), lowerLimit);
```
Comment (mark-prins) [inline src/widgets/StepperInput.js]: very nifty!

File: src/widgets/StepperInput.js [hunk 2]
```diff
@@ -23,26 +26,33 @@ import { AddIcon, MinusIcon } from './icons';
...
   const onUpdate = newValue => {
-    currentValue.current = newValue;
-    onChangeText(newValue);
+    const updateValue = parseNumber(newValue);
+    currentValue.current = updateValue;
+    setCurrentValueState(updateValue);
+    onChangeText(updateValue);
   };
 
   const decrementValue = () => {
```
Comment (mark-prins) [inline src/widgets/StepperInput.js]: or.. 

```
const decrementValue = () => onUpdate(currentValue.current - currentAdjustmentAmount.current);

const incrementValue = () => onUpdate(currentValue.current + currentAdjustmentAmount.current);
```

File: src/widgets/StepperInput.js [hunk 3]
```diff
@@ -15,6 +15,9 @@ import { AddIcon, MinusIcon } from './icons';
  * The buttons can also be held, increase by one every 50ms. The increment amount
  * then grows exponentially every 500ms, reseting to one after releasing.
  *
+ * Component uses state to update itself, and a ref for storing the current value. This
+ * is a bit of a hack, but is much more responsive to the rapid updates of the input.
```
Comment (mark-prins) [inline src/widgets/StepperInput.js]: maybe just me, but I don't see it as that hacky. For a component like this it is sensible to do the rendering based on local state. And my understanding of the `useRef` is that it is perfect for this sort of thing.

File: src/widgets/StepperInput.js (hunk 2 above)
Comment (joshxg) [reply src/widgets/StepperInput.js]: nice thanks #1821

File: src/widgets/StepperInput.js (hunk 3 above)
Comment (joshxg) [reply src/widgets/StepperInput.js]: I guess more of a creative use of tools? 🤣

**Pull request merged**